bosh/proxy/socks5_proxy.go

```diff
@@ -0,0 +1,113 @@
+package proxy
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+
+	socks5 "github.com/armon/go-socks5"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/net/context"
+)
+
+var netListen = net.Listen
+
+//go:generate counterfeiter . Proxy
+type Proxy interface {
+	Start(string, string) error
+	Addr() (string, error)
+}
+
+type Socks5Proxy struct {
+	hostKeyGetter KeyGetter
+	port          int
+	started       bool
+}
+
+func NewSocks5Proxy(hostKeyGetter KeyGetter) *Socks5Proxy {
+	return &Socks5Proxy{
+		hostKeyGetter: hostKeyGetter,
+		started:       false,
+	}
+}
+
+func (s *Socks5Proxy) Start(key, url string) error {
+	if s.started {
+		return nil
+	}
+
+	signer, err := ssh.ParsePrivateKey([]byte(key))
+	if err != nil {
+		return fmt.Errorf("parse private key: %s", err)
+	}
+
+	hostKey, err := s.hostKeyGetter.Get(key, url)
+	if err != nil {
+		return fmt.Errorf("get host key: %s", err)
+	}
+
+	clientConfig := &ssh.ClientConfig{
+		User: "jumpbox",
```

Comment:
Should this be made configurable? Then this would work for non-bbl envs as well. This would be tied to the other usage in the host_key_getter.